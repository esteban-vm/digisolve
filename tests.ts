import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
