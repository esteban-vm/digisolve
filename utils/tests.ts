import type { ReactElement } from 'react'
import { create, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON()
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
