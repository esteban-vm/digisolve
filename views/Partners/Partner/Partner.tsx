import type { Component, PropsWithImage } from '@/types'
import Image from 'next/image'
import { styled } from '@/styles'

export type PartnerProps = PropsWithImage<{ text: string }>

const PartnerComponent: Component<PartnerProps> = ({ id, text, image, ...rest }) => {
  return (
    <article aria-labelledby={id} {...rest}>
      <Image alt={text} id={id} {...image} />
    </article>
  )
}

const Partner = styled(PartnerComponent)`
  img {
    width: 65%;
    height: auto;
  }
`

export default Partner
