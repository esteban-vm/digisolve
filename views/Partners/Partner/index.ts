export { default /* type PartnerProps */ } from './Partner'
