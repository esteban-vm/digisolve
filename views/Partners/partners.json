[
  {
    "desc": "Partner",
    "image": {
      "src": "/img/brand-logo-1.png",
      "height": 500,
      "width": 500
    }
  },
  {
    "desc": "Partner",
    "image": {
      "src": "/img/brand-logo-2.png",
      "height": 500,
      "width": 500
    }
  },
  {
    "desc": "Partner",
    "image": {
      "src": "/img/brand-logo-3.png",
      "height": 500,
      "width": 500
    }
  },
  {
    "desc": "Partner",
    "image": {
      "src": "/img/brand-logo-4.png",
      "height": 500,
      "width": 500
    }
  },
  {
    "desc": "Partner",
    "image": {
      "src": "/img/brand-logo-5.png",
      "height": 500,
      "width": 500
    }
  },
  {
    "desc": "Partner",
    "image": {
      "src": "/img/brand-logo-6.png",
      "height": 500,
      "width": 500
    }
  },
  {
    "desc": "Partner",
    "image": {
      "src": "/img/brand-logo-7.png",
      "height": 500,
      "width": 500
    }
  },
  {
    "desc": "Partner",
    "image": {
      "src": "/img/brand-logo-8.png",
      "height": 500,
      "width": 500
    }
  }
]
