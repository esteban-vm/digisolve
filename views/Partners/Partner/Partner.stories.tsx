import { type Meta, type StoryObj, within, userEvent } from '@/stories'
import { Grid } from '@/components'
import { styled } from '@/styles'
import { partners } from '../Partners'
import Partner from './Partner'

const Wrapper = styled.div`
  text-align: center;
  width: 50vw;

  article {
    background-color: var(--color-white);
  }
`

export default {
  argTypes: {
    src: {
      control: false,
    },
    alt: {
      control: false,
    },
  },
  component: Partner,
  title: 'Views/Partners/Partners',
  decorators: [
    (Story, context) => {
      return (
        <Wrapper>
          <Grid.Row>
            <Grid.Col isOneThird>&nbsp;</Grid.Col>
            <Grid.Col isOneThird>
              <Story {...context} />
            </Grid.Col>
            <Grid.Col isOneThird>&nbsp;</Grid.Col>
          </Grid.Row>
        </Wrapper>
      )
    },
  ],
  parameters: {
    layout: 'centered',
  },
  play({ canvasElement }) {
    const canvas = within(canvasElement)
    const logo = canvas.getByRole('img')
    userEvent.hover(logo)
  },
} as Meta<typeof Partner>

const Examples: StoryObj<typeof Partner>[] = []
for (const partner of partners) Examples.push({ args: partner })

export const Example1 = Examples[0]
export const Example2 = Examples[1]
export const Example3 = Examples[2]
export const Example4 = Examples[3]
export const Example5 = Examples[4]
export const Example6 = Examples[5]
export const Example7 = Examples[6]
export const Example8 = Examples[7]
