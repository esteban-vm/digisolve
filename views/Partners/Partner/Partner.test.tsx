import { render, cleanup, screen, create } from '@/tests'
import partners from '../Partners.data'
import Partner from './Partner'

describe('ðŸ§ª PARTNER:', () => {
  const [testPartner] = partners

  describe('display tests:', () => {
    beforeEach(() => {
      render(<Partner {...testPartner} />)
    })

    afterEach(cleanup)

    it('should be accessible', () => {
      const article = screen.getByRole('article', { name: testPartner.text })
      expect(article).toBeInTheDocument()
    })

    it('the logo should be displayed', () => {
      const logo = screen.getByRole('img', { name: testPartner.text })
      expect(logo).toBeInTheDocument()
      expect(logo).toBeVisible()
    })
  })

  describe('style tests:', () => {
    it('should render with correct styles', () => {
      const tree = create(<Partner {...testPartner} />)
      expect(tree).toMatchSnapshot()
    })
  })
})
