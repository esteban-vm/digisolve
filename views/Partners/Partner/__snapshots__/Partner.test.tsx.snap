// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª PARTNER: style tests: should render with correct styles 1`] = `
.emotion-0 img {
  width: 65%;
  height: auto;
}

<article
  aria-labelledby="473a9e41-bb65-4542-957d-62ed0bcfd039"
  className="emotion-0"
>
  <img
    alt="Partner"
    data-nimg="1"
    decoding="async"
    height={500}
    id="473a9e41-bb65-4542-957d-62ed0bcfd039"
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fimg%2Fbrand-logo-1.png&w=1080&q=75"
    srcSet="/_next/image?url=%2Fimg%2Fbrand-logo-1.png&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fbrand-logo-1.png&w=1080&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    width={500}
  />
</article>
`;
